### ISSUE TITLE
[BUG]--cross_entropy_fusion_impl te results in grad_norm 0

### ISSUE THREAD

**Issue by Jintao-Huang** [NVIDIA/Megatron-LM#1750]
After passing through this function:
https://github.com/NVIDIA/TransformerEngine/blob/6ba98d439190901fec85c2ae3c2cea235d9f2196/transformer_engine/pytorch/triton/cross_entropy.py#L359-L365

_input becomes all zeros.

<img width="791" height="91" alt="Image" src="https://github.com/user-attachments/assets/358fa899-63a2-431e-8afc-721f23869198" />

**Comment by Skylion007** [comment]
What version of Transformer Engine, TE was fixed in the latest version of transformer engine, 2.5 (which is not in NGC 25.06 sadly).

**Comment by casper-hansen** [comment]
@Skylion007 I am getting this too (latest TE 2.5.0). Please research and fix this issue (brown is te, green is native)

<img width="727" height="313" alt="Image" src="https://github.com/user-attachments/assets/c5ad444c-1aa9-4288-9b08-e9cd1651126d" />

With this gradient norm bug, my model does not convergence during training and actually completely diverges and causes training not to work.

<img width="722" height="309" alt="Image" src="https://github.com/user-attachments/assets/9a84ca4c-bed4-46bd-afe2-a3540ba90806" />

**Comment by casper-hansen** [comment]
fixed in https://github.com/NVIDIA/TransformerEngine/pull/2139

**Comment by sbhavani** [comment]
@ksivaman
